### PR TITLE
Jenkins/code quality

### PR DESCRIPTION
**Pull request by stevegal**
Add in a collector for junit/pmd/findbugs/checkstyle for those who don't use sonar.

### CODE REVIEW

**Comment by stevegal** [comment]
I'm currently working on adding in publish into the pipeline scripts. Do you want both in one PR? I was going to follow up if this one got in. Thoughts?

**Comment by stevegal** [comment]
doh! forgot to set the extra data so the results appear in the UI. I'll add this over the weekend.

**Comment by xiaods** [comment]
please rebase

**Comment by stevegal** [comment]
okay. I'll do that soon.

**Comment by stevegal** [comment]
done

**Comment by stevegal** [comment]
shall I close this PR? Any chance of it getting merged?

**Comment by satishc1** [comment]
@stevegal Will look into this one next week. All contributions are welcome. Thanks.

**Comment by satishc1** [comment]
@stevegal Thanks for the contribution. To maintain the naming convention, do you mind changing the directory name from 
```
collectors/build/jenkins-codequality-collector/
```
to
```
collectors/build/jenkins-codequality/
```

The artifactId in the pom is correct and needs to be 
```
<artifactId>jenkins-codequality-collector</artifactId>
```

File: collectors/build/jenkins-codequality-collector/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+To package the collector into an executable JAR file, run:
+```
+mvn install
+```
+
+Copy this file to your server and launch it using :
+```
+java -JAR jenkins-codequlaity-collector.jar
+```
```
Comment (satishc1) [inline collectors/build/jenkins-codequality-collector/README.md]: @stevegal typo here.

File: collectors/build/jenkins-codequality-collector/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+FROM docker.io/java:openjdk-8-jdk
```
Comment (satishc1) [inline collectors/build/jenkins-codequality-collector/docker/Dockerfile]: @stevegal Please change to 
```
FROM docker.io/openjdk:8-jre
```

File: collectors/build/jenkins-codequality-collector/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+MAINTAINER Hygieia@capitalone.com
+
+RUN \
+  mkdir /hygieia
+
+COPY *.jar /hygieia/
+COPY bamboo-build-properties-builder.sh /hygieia/
```
Comment (satishc1) [inline collectors/build/jenkins-codequality-collector/docker/Dockerfile]: @stevegal Please change to reflect the correct builder.sh
```
COPY jenkins-codequality-properties-builder.sh /hygieia/
```

File: pom.xml [hunk 4]
```diff
@@ -172,6 +186,7 @@
         <module>collectors/build/bamboo</module>
         <module>collectors/build/jenkins</module>
         <module>collectors/build/jenkins-cucumber</module>
+        <module>collectors/build/jenkins-codequality-collector</module>
         <module>collectors/build/sonar</module>
```
Comment (satishc1) [inline pom.xml]: @stevegal Change to 
```
<module>collectors/build/jenkins-codequality</module>
```

File: pom.xml [hunk 5]
```diff
@@ -215,6 +230,7 @@
         <module>collectors/build/bamboo</module>
         <module>collectors/build/jenkins</module>
         <module>collectors/build/jenkins-cucumber</module>
+        <module>collectors/build/jenkins-codequality-collector</module>
```
Comment (satishc1) [inline pom.xml]: @stevegal Change to 
```
<module>collectors/build/jenkins-codequality</module>
```

**Comment by stevegal** [comment]
@satishc1 thanks for the comments. I was away from my keyboard for a few days, hence the delay in updating.

**Comment by satishc1** [comment]
@stevegal No problem. Thanks for the contribution. Will try to merge it this week.

**Comment by satishc1** [comment]
@tabladrum LGTM

**Pull request merged**